src/libraries/System.Reflection.Metadata/tests/PortableExecutable/PEHeadersTests.cs

```diff
@@ -50,6 +50,13 @@ public void Ctor_Streams()
             Assert.Equal(0, s.Position);
         }
 
+        [Fact]
+        public void Ctor_InvalidFile()
```

Comment:
What case is this catching from above, and related question is what coverage do we have on the new checks?

Reply:
It catches the "size of optional header is nonzero" case. I can add more tests.